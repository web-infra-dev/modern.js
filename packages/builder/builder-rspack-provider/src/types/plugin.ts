import type {
  PluginStore,
  DefaultBuilderPluginAPI,
  BuilderPlugin as BaseBuilderPlugin,
} from '@modern-js/builder-shared';
import type { BuilderConfig, NormalizedConfig } from './config';
import type {
  ModifyRspackConfigFn,
  ModifyBuilderConfigFn,
  OnAfterCreateCompilerFn,
  OnBeforeCreateCompilerFn,
} from './hooks';
import type { RspackConfig } from './rspack';

export interface BuilderPluginAPI
  extends DefaultBuilderPluginAPI<
    BuilderConfig,
    NormalizedConfig,
    RspackConfig
  > {
  isPluginExists: PluginStore['isPluginExists'];
  /**
   * Get the relative paths of generated HTML files.
   * The key is entry name and the value is path.
   */
  getHTMLPaths: () => Record<string, string>;

  // Hooks
  onAfterCreateCompiler: (fn: OnAfterCreateCompilerFn) => void;
  onBeforeCreateCompiler: (fn: OnBeforeCreateCompilerFn) => void;

  // Modifiers
  modifyRspackConfig: (fn: ModifyRspackConfigFn) => void;
  modifyBuilderConfig: (fn: ModifyBuilderConfigFn) => void;
}

export type BuilderPlugin = BaseBuilderPlugin<BuilderPluginAPI>;
