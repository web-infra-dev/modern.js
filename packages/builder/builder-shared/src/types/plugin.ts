import type {
  OnExitFn,
  OnAfterBuildFn,
  OnBeforeBuildFn,
  OnDevCompileDoneFn,
  OnAfterStartDevServerFn,
  OnBeforeStartDevServerFn,
  OnAfterCreateCompilerFn,
  OnBeforeCreateCompilerFn,
  ModifyBuilderConfigFn,
  ModifyBundlerChainFn,
} from './hooks';
import { BuilderContext } from './context';
import { SharedBuilderConfig, SharedNormalizedConfig } from './config';
import { PromiseOrNot } from './utils';

export type PluginStore = {
  readonly plugins: BuilderPlugin[];
  addPlugins: (plugins: BuilderPlugin[], options?: { before?: string }) => void;
  removePlugins: (pluginNames: string[]) => void;
  isPluginExists: (pluginName: string) => boolean;
  /** The plugin API. */
  pluginAPI?: DefaultBuilderPluginAPI;
};

export type BuilderPlugin<API = any> = {
  name: string;
  setup: (api: API) => PromiseOrNot<void>;
  pre?: string[];
  post?: string[];
  remove?: string[];
};

type PluginsFn<T = void> = T extends void
  ? () => Promise<BuilderPlugin>
  : (arg: T) => Promise<BuilderPlugin>;

export type Plugins = {
  cleanOutput: PluginsFn;
  startUrl: PluginsFn;
  fileSize: PluginsFn;
  devtool: PluginsFn;
  target: PluginsFn;
  entry: PluginsFn;
  cache: PluginsFn;
  yaml: PluginsFn;
  toml: PluginsFn;
  splitChunks: PluginsFn;
  inlineChunk: PluginsFn;
  bundleAnalyzer: PluginsFn;
  assetsRetry: PluginsFn;
  font: PluginsFn;
  media: PluginsFn;
  image: PluginsFn;
  svg: PluginsFn;
  html: PluginsFn;
  antd: PluginsFn;
  arco: PluginsFn;
  tsChecker: PluginsFn;
  checkSyntax: PluginsFn;
  rem: PluginsFn;
  wasm: PluginsFn;
  moment: PluginsFn;
  nodeAddons: PluginsFn;
  externals: PluginsFn;
  sourceBuild: PluginsFn;
  networkPerformance: PluginsFn;
  preloadOrPrefetch: PluginsFn;
  performance: PluginsFn;
  define: PluginsFn;
};

/**
 * Define a generic builder plugin API that provider can extend as needed.
 */
export type DefaultBuilderPluginAPI<
  Config extends Record<string, any> = Record<string, any>,
  NormalizedConfig extends Record<string, any> = Record<string, any>,
  BundlerConfig = unknown,
  Compiler = unknown,
> = {
  context: Readonly<BuilderContext>;
  isPluginExists: PluginStore['isPluginExists'];

  onExit: (fn: OnExitFn) => void;
  onAfterBuild: (fn: OnAfterBuildFn) => void;
  onBeforeBuild: (fn: OnBeforeBuildFn<BundlerConfig>) => void;
  onDevCompileDone: (fn: OnDevCompileDoneFn) => void;
  onAfterStartDevServer: (fn: OnAfterStartDevServerFn) => void;
  onBeforeStartDevServer: (fn: OnBeforeStartDevServerFn) => void;
  onAfterCreateCompiler: (fn: OnAfterCreateCompilerFn<Compiler>) => void;
  onBeforeCreateCompiler: (fn: OnBeforeCreateCompilerFn<BundlerConfig>) => void;

  /**
   * Get the relative paths of generated HTML files.
   * The key is entry name and the value is path.
   */
  getHTMLPaths: () => Record<string, string>;
  getBuilderConfig: () => Readonly<Config>;
  getNormalizedConfig: () => NormalizedConfig;

  modifyBuilderConfig: (fn: ModifyBuilderConfigFn<Config>) => void;
  modifyBundlerChain: (fn: ModifyBundlerChainFn) => void;
};

export type SharedBuilderPluginAPI = DefaultBuilderPluginAPI<
  SharedBuilderConfig,
  SharedNormalizedConfig
>;

export type DefaultBuilderPlugin = BuilderPlugin<SharedBuilderPluginAPI>;
