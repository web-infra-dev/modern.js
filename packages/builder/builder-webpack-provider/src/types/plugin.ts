import type {
  BuilderContext,
  PluginStore,
  BuilderPlugin as BaseBuilderPlugin,
} from '@modern-js/builder-shared';
import type { BuilderConfig, NormalizedConfig } from './config';
import type {
  OnExitFn,
  OnAfterBuildFn,
  OnBeforeBuildFn,
  OnDevCompileDoneFn,
  ModifyWebpackChainFn,
  ModifyWebpackConfigFn,
  ModifyBuilderConfigFn,
  OnAfterCreateCompilerFn,
  OnBeforeCreateCompilerFn,
  OnAfterStartDevServerFn,
  OnBeforeStartDevServerFn,
} from './hooks';

export type BuilderPluginAPI = {
  context: Readonly<BuilderContext>;
  isPluginExists: PluginStore['isPluginExists'];
  /**
   * Get the relative paths of generated HTML files.
   * The key is entry name and the value is path.
   */
  getHTMLPaths: () => Record<string, string>;
  getBuilderConfig: () => Readonly<BuilderConfig>;
  getNormalizedConfig: () => Readonly<NormalizedConfig>;

  // Hooks
  onExit: (fn: OnExitFn) => void;
  onAfterBuild: (fn: OnAfterBuildFn) => void;
  onBeforeBuild: (fn: OnBeforeBuildFn) => void;
  onDevCompileDone: (fn: OnDevCompileDoneFn) => void;
  onAfterCreateCompiler: (fn: OnAfterCreateCompilerFn) => void;
  onBeforeCreateCompiler: (fn: OnBeforeCreateCompilerFn) => void;
  onAfterStartDevServer: (fn: OnAfterStartDevServerFn) => void;
  onBeforeStartDevServer: (fn: OnBeforeStartDevServerFn) => void;

  // Modifiers
  modifyWebpackChain: (fn: ModifyWebpackChainFn) => void;
  modifyWebpackConfig: (fn: ModifyWebpackConfigFn) => void;
  modifyBuilderConfig: (fn: ModifyBuilderConfigFn) => void;
};

export type BuilderPlugin = BaseBuilderPlugin<BuilderPluginAPI>;
