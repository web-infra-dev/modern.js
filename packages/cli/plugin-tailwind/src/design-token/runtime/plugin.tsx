import type { Plugin } from '@modern-js/runtime';
import hoistNonReactStatics from 'hoist-non-react-statics';
import React, { useContext } from 'react';

export const DesignTokenContext = React.createContext<any>({});

export const useDesignTokens = (): IDesignTokens =>
  useContext<IDesignTokens>(DesignTokenContext);

export default (
  options: {
    token?: Record<string, any>;
    useStyledComponentsThemeProvider?: boolean;
    useDesignTokenContext?: boolean;
  } = {},
): Plugin => ({
  name: '@modern-js/plugin-design-token',

  setup: () => ({
    hoc({ App }, next) {
      const DesignTokenAppWrapper = (props: any) => {
        const {
          token = {},
          useStyledComponentsThemeProvider = false,
          useDesignTokenContext = false,
        } = options;
        if (useStyledComponentsThemeProvider && useDesignTokenContext) {
          const { ThemeProvider } = require('@modern-js/runtime/styled');
          return (
            <ThemeProvider theme={token}>
              <DesignTokenContext.Provider value={token}>
                <App {...props} />
              </DesignTokenContext.Provider>
            </ThemeProvider>
          );
        } else if (useStyledComponentsThemeProvider) {
          const { ThemeProvider } = require('@modern-js/runtime/styled');
          return (
            <ThemeProvider theme={token}>
              <App {...props} />
            </ThemeProvider>
          );
        } else if (useDesignTokenContext) {
          return (
            <DesignTokenContext.Provider value={token}>
              <App {...props} />
            </DesignTokenContext.Provider>
          );
        } else {
          return <App {...props} />;
        }
      };

      return next({
        App: hoistNonReactStatics(DesignTokenAppWrapper, App),
      });
    },
  }),
});
