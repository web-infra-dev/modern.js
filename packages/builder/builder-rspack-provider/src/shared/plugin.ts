import { BuilderPlugin } from '../types';
import { awaitableGetter, Plugins } from '@modern-js/builder-shared';

export const applyMinimalPlugins = (plugins: Plugins) =>
  awaitableGetter<BuilderPlugin>([
    import('../plugins/basic').then(m => m.builderPluginBasic()),
    plugins.entry(),
    plugins.cache(),
    plugins.target(),
    import('../plugins/output').then(m => m.builderPluginOutput()),
    plugins.devtool(),
    import('../plugins/resolve').then(m => m.builderPluginResolve()),
    plugins.fileSize(),
    // should before the html plugin
    plugins.cleanOutput(),
    import('../plugins/font').then(m => m.builderPluginFont()),
    import('../plugins/image').then(m => m.builderPluginImage()),
    import('../plugins/media').then(m => m.builderPluginMedia()),
    import('../plugins/html').then(m => m.builderPluginHtml()),
    import('../plugins/define').then(m => m.builderPluginDefine()),
    import('../plugins/css').then(m => m.builderPluginCss()),
    import('../plugins/less').then(m => m.builderPluginLess()),
    import('../plugins/sass').then(m => m.builderPluginSass()),
  ]);

export const applyDefaultPlugins = (plugins: Plugins) =>
  awaitableGetter<BuilderPlugin>([
    ...applyMinimalPlugins(plugins).promises,
    import('../plugins/hmr').then(m => m.builderPluginHMR()),
    import('../plugins/progress').then(m => m.builderPluginProgress()),
    import('../plugins/react').then(m => m.builderPluginReact()),
    import('../plugins/externals').then(m => m.builderPluginExternals()),
    plugins.toml(),
    plugins.yaml(),
    plugins.startUrl(),
  ]);
