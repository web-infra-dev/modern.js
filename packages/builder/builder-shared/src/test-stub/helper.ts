import { Plugins } from '../types';

// eslint-disable-next-line @typescript-eslint/no-empty-function
const setup = () => {};

const genMockPlugin = (name: string) => () =>
  Promise.resolve({
    name,
    setup,
  });

export const mockBuilderPlugins: Plugins = {
  cleanOutput: genMockPlugin('builder-plugin-clean-output'),
  startUrl: genMockPlugin('builder-plugin-start-url'),
  fileSize: genMockPlugin('builder-plugin-file-size'),
  target: genMockPlugin('builder-plugin-target'),
  devtool: genMockPlugin('builder-plugin-devtool'),
  entry: genMockPlugin('builder-plugin-entry'),
  cache: genMockPlugin('builder-plugin-cache'),
  yaml: genMockPlugin('builder-plugin-yaml'),
  toml: genMockPlugin('builder-plugin-toml'),
  splitChunks: genMockPlugin('builder-plugin-split-chunks'),
  inlineChunk: genMockPlugin('builder-plugin-inline-chunk'),
  bundleAnalyzer: genMockPlugin('builder-plugin-bundle-analyzer'),
  assetsRetry: genMockPlugin('builder-plugin-assets-retry'),
  font: genMockPlugin('builder-plugin-font'),
  media: genMockPlugin('builder-plugin-media'),
  image: genMockPlugin('builder-plugin-image'),
  svg: genMockPlugin('builder-plugin-svg'),
  html: genMockPlugin('builder-plugin-html'),
  antd: genMockPlugin('antd'),
  arco: genMockPlugin('arco'),
  tsChecker: genMockPlugin('builder-plugin-ts-checker'),
  checkSyntax: genMockPlugin('builder-plugin-check-syntax'),
  rem: genMockPlugin('builder-plugin-rem'),
  wasm: genMockPlugin('builder-plugin-wasm'),
  moment: genMockPlugin('builder-plugin-moment'),
  nodeAddons: genMockPlugin('builder-plugin-node-addons'),
  externals: genMockPlugin('builder-plugin-externals'),
  sourceBuild: genMockPlugin('builder-plugin-source-build'),
  networkPerformance: genMockPlugin('builder-plugin-network-performance'),
  preloadOrPrefetch: genMockPlugin('builder-plugin-preload-or-prefetch'),
  performance: genMockPlugin('builder-plugin-performance'),
  define: genMockPlugin('builder-plugin-define'),
};
