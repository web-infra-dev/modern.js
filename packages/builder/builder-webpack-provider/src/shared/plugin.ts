import { BuilderPlugin } from '../types';
import { awaitableGetter, Plugins } from '@modern-js/builder-shared';

export const applyMinimalPlugins = (plugins: Plugins) =>
  awaitableGetter<BuilderPlugin>([
    import('../plugins/basic').then(m => m.PluginBasic()),
    plugins.entry(),
    import('../plugins/cache').then(m => m.PluginCache()),
    plugins.target(),
    import('../plugins/output').then(m => m.PluginOutput()),
    plugins.devtool(),
    import('../plugins/resolve').then(m => m.PluginResolve()),
  ]);

export const applyBasicPlugins = (plugins: Plugins) =>
  awaitableGetter<BuilderPlugin>([
    ...applyMinimalPlugins(plugins).promises,
    import('../plugins/copy').then(m => m.PluginCopy()),
    import('../plugins/html').then(m => m.PluginHtml()),
    import('../plugins/image').then(m => m.PluginImage()),
    import('../plugins/define').then(m => m.PluginDefine()),
    import('../plugins/tsLoader').then(m => m.PluginTsLoader()),
    import('../plugins/babel').then(m => m.PluginBabel()),
    import('../plugins/css').then(m => m.PluginCss()),
    import('../plugins/sass').then(m => m.PluginSass()),
    import('../plugins/less').then(m => m.PluginLess()),
    import('../plugins/react').then(m => m.PluginReact()),
  ]);

export const applyDefaultPlugins = (plugins: Plugins) =>
  awaitableGetter<BuilderPlugin>([
    ...applyMinimalPlugins(plugins).promises,
    plugins.fileSize(),
    plugins.cleanOutput(),
    import('../plugins/hmr').then(m => m.PluginHMR()),
    import('../plugins/svg').then(m => m.PluginSvg()),
    import('../plugins/pug').then(m => m.PluginPug()),
    import('../plugins/copy').then(m => m.PluginCopy()),
    import('../plugins/font').then(m => m.PluginFont()),
    import('../plugins/html').then(m => m.PluginHtml()),
    import('../plugins/image').then(m => m.PluginImage()),
    import('../plugins/media').then(m => m.PluginMedia()),
    import('../plugins/moment').then(m => m.PluginMoment()),
    import('../plugins/define').then(m => m.PluginDefine()),
    import('../plugins/progress').then(m => m.PluginProgress()),
    import('../plugins/minimize').then(m => m.PluginMinimize()),
    import('../plugins/manifest').then(m => m.PluginManifest()),
    import('../plugins/moduleScopes').then(m => m.PluginModuleScopes()),
    import('../plugins/tsLoader').then(m => m.PluginTsLoader()),
    import('../plugins/babel').then(m => m.PluginBabel()),
    import('../plugins/tsChecker').then(m => m.PluginTsChecker()),
    import('../plugins/css').then(m => m.PluginCss()),
    import('../plugins/sass').then(m => m.PluginSass()),
    import('../plugins/less').then(m => m.PluginLess()),
    import('../plugins/rem').then(m => m.PluginRem()),
    import('../plugins/react').then(m => m.PluginReact()),
    import('../plugins/bundleAnalyzer').then(m => m.PluginBundleAnalyzer()),
    import('../plugins/toml').then(m => m.PluginToml()),
    import('../plugins/yaml').then(m => m.PluginYaml()),
    import('../plugins/splitChunks').then(m => m.PluginSplitChunks()),
    import('../plugins/inspector').then(m => m.PluginInspector()),
    import('../plugins/sri').then(m => m.PluginSRI()),
    plugins.startUrl(),
    import('../plugins/inlineChunk').then(m => m.PluginInlineChunk()),
    import('../plugins/assetsRetry').then(m => m.PluginAssetsRetry()),
    import('../plugins/externals').then(m => m.PluginExternals()),
    import('../plugins/performance').then(m => m.PluginPerformance()),
    import('../plugins/lazyCompilation').then(m => m.PluginLazyCompilation()),
    import('../plugins/fallback').then(m => m.PluginFallback()), // fallback should be the last plugin
  ]);
