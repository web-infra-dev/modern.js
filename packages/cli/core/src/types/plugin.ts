import { AsyncSetup, PluginOptions } from '@modern-js/plugin';
import { BaseHooks } from './hooks';
import { BasePluginAPI } from './pluginAPI';

export type { InternalPlugins } from '@modern-js/types';

/** Plugin options of a cli plugin. */
export type CliPlugin<
  Extends extends {
    hooks?: ExtendHooks;
    userConfig?: ExtendUserConfig;
    normalizedConfig?: ExtendNormalizedConfig;
    // eslint-disable-next-line @typescript-eslint/ban-types
  } = {},
  // eslint-disable-next-line @typescript-eslint/ban-types
  ExtendHooks extends Record<string, any> = {},
  // eslint-disable-next-line @typescript-eslint/ban-types
  ExtendUserConfig extends Record<string, any> = {},
  // eslint-disable-next-line @typescript-eslint/ban-types
  ExtendNormalizedConfig extends Record<string, any> = {},
> = PluginOptions<
  BaseHooks<Extends>,
  AsyncSetup<BaseHooks<Extends> & Extends['hooks'], BasePluginAPI<Extends>>,
  Extends['hooks'],
  // eslint-disable-next-line @typescript-eslint/ban-types
  BaseHooks<Extends> | BaseHooks<{}>,
  | AsyncSetup<BaseHooks<Extends> & Extends['hooks'], BasePluginAPI<Extends>>
  // eslint-disable-next-line @typescript-eslint/ban-types
  | AsyncSetup<BaseHooks<{}>, BasePluginAPI<{}>>
>;

export type PluginItem = string | [string, any];

/**
 * @deprecated
 * Using NewPluginConfig instead.
 */
export type OldPluginConfig = Array<PluginItem>;

export type NewPluginConfig<
  PluginTypes extends {
    hooks?: Record<string, any>;
    userConfig?: Record<string, any>;
    normalizedConfig?: Record<string, any>;
  },
> = CliPlugin<PluginTypes>[];

export type PluginConfig<
  PluginTypes extends {
    hooks?: Record<string, any>;
    userConfig?: Record<string, any>;
    normalizedConfig?: Record<string, any>;
    // eslint-disable-next-line @typescript-eslint/ban-types
  } = {},
> = OldPluginConfig | NewPluginConfig<PluginTypes>;
