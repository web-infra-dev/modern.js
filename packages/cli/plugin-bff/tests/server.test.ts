import path from 'path';
import {
  PluginManager,
  ServerPlugin,
  createContext,
} from '@modern-js/server-core';
import plugin from '../src/server';
import './helper';

// eslint-disable-next-line @typescript-eslint/no-empty-function
export const noop = () => {};

const pwd = path.resolve(__dirname, './fixtures/function');

function createRunner(plugins?: ServerPlugin[]) {
  const appContext = createContext<any>({});
  const pluginManager = new PluginManager({
    appContext,
    cliConfig: {
      html: {},
      output: {},
      source: {},
      tools: {},
      server: {},
      runtime: {},
      bff: {},
      dev: {},
      security: {},
    },
  });

  plugins && pluginManager.addPlugins(plugins);

  return pluginManager.init();
}

describe('bff server plugin', () => {
  describe('prepareApiServer', () => {
    it('should work well', async () => {
      let apiHandlerInfos = null;
      const mockApiPlugin: ServerPlugin = {
        name: 'mock-api',

        setup(api) {
          return {
            prepareApiServer(props, next) {
              const appContext = api.useAppContext();
              // eslint-disable-next-line prefer-destructuring
              apiHandlerInfos = appContext.apiHandlerInfos;
              return next(props);
            },
          };
        },
      };

      const runner = await createRunner([plugin(), mockApiPlugin]);

      await runner.prepareApiServer(
        {
          pwd,
          mode: 'function',
          prefix: '/',
        },
        { onLast: () => noop },
      );

      expect(apiHandlerInfos).toMatchSnapshot();
    });

    it('should work well with prefix', async () => {
      let apiHandlerInfos = null;

      const mockApiPlugin: ServerPlugin = {
        name: 'mock-api',

        setup(api) {
          return {
            prepareApiServer(props, next) {
              const appContext = api.useAppContext();
              // eslint-disable-next-line prefer-destructuring
              apiHandlerInfos = appContext.apiHandlerInfos;
              return next(props);
            },
          };
        },
      };

      const runner = await createRunner([plugin(), mockApiPlugin]);

      await runner.prepareApiServer(
        { pwd, mode: 'function', prefix: '/api' },
        { onLast: () => noop },
      );
      expect(apiHandlerInfos).toMatchSnapshot();
    });
  });
});
