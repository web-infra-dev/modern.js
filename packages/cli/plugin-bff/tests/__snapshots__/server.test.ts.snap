// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bff server plugin prepareApiServer should work well 1`] = `
[
  {
    "filename": "<ROOT>/fixtures/function/api/hello.ts",
    "handler": [Function],
    "httpMethod": "GET",
    "name": "get",
    "routeName": "/hello",
    "routePath": "/hello",
  },
  {
    "filename": "<ROOT>/fixtures/function/api/upload.ts",
    "handler": [Function],
    "httpMethod": "GET",
    "name": "get",
    "routeName": "/upload",
    "routePath": "/upload",
  },
  {
    "filename": "<ROOT>/fixtures/function/api/user/[id].ts",
    "handler": [Function],
    "httpMethod": "GET",
    "name": "get",
    "routeName": "/user/:id",
    "routePath": "/user/:id",
  },
]
`;

exports[`bff server plugin prepareApiServer should work well with prefix 1`] = `
[
  {
    "filename": "<ROOT>/fixtures/function/api/hello.ts",
    "handler": [Function],
    "httpMethod": "GET",
    "name": "get",
    "routeName": "/hello",
    "routePath": "/api/hello",
  },
  {
    "filename": "<ROOT>/fixtures/function/api/upload.ts",
    "handler": [Function],
    "httpMethod": "GET",
    "name": "get",
    "routeName": "/upload",
    "routePath": "/api/upload",
  },
  {
    "filename": "<ROOT>/fixtures/function/api/user/[id].ts",
    "handler": [Function],
    "httpMethod": "GET",
    "name": "get",
    "routeName": "/user/:id",
    "routePath": "/api/user/:id",
  },
]
`;
